app/validators/blocks/list_collector_validator.py

```diff
@@ -96,53 +111,45 @@ def validate_other_list_collectors(self):
             - duplicate answer id in add, edit, or remove block for other different-named list collectors
         """
-        list_blocks_to_validate = ["add_block", "edit_block", "remove_block"]
-
         list_name = self.block["for_list"]
         list_blocks_answer_ids = {
-            list_block: self.questionnaire_schema.get_all_answer_ids(
-                self.block[list_block]["id"]
+            child_block: self.questionnaire_schema.get_all_answer_ids(
+                self.block[child_block]["id"]
             )
-            for list_block in list_blocks_to_validate
+            for child_block in self.LIST_CHILD_BLOCKS_TO_VALIDATE
         }
 
         other_list_collectors = self.questionnaire_schema.get_other_blocks(
             self.block["id"], type="ListCollector"
         )
 
         for other_list_collector in other_list_collectors:
-            other_list_name = other_list_collector["for_list"]
-            are_list_names_matching = other_list_name == list_name
-
             other_list_blocks_answer_ids = {
-                other_list_block: self.questionnaire_schema.get_all_answer_ids(
-                    other_list_collector[other_list_block]["id"]
+                child_block: self.questionnaire_schema.get_all_answer_ids(
+                    other_list_collector[child_block]["id"]
                 )
-                for other_list_block in list_blocks_to_validate
+                for child_block in self.LIST_CHILD_BLOCKS_TO_VALIDATE
             }
 
-            if are_list_names_matching and list_blocks_answer_ids[
-                "add_block"
-            ].symmetric_difference(other_list_blocks_answer_ids["add_block"]):
-                self.add_error(
-                    self.NON_UNIQUE_ANSWER_ID_FOR_SAME_LIST_COLLECTOR_ADD_BLOCK,
-                    list_name=list_name,
-                )
-
-            # Validate other different list collector
-            if not are_list_names_matching:
-                for list_block_to_validate in list_blocks_to_validate:
-                    has_duplicate_answer_ids = any(
-                        answer_id
-                        in other_list_blocks_answer_ids[list_block_to_validate]
-                        for answer_id in list_blocks_answer_ids[list_block_to_validate]
+            if list_name == other_list_collector["for_list"]:
+                if list_blocks_answer_ids["add_block"].symmetric_difference(
+                    other_list_blocks_answer_ids["add_block"]
+                ):
+                    self.add_error(
+                        self.DIFFERENT_LIST_COLLECTOR_ADD_BLOCKS_FOR_SAME_LIST,
+                        list_name=list_name,
                     )
-                    if has_duplicate_answer_ids:
+            else:
+                for child_block in self.LIST_CHILD_BLOCKS_TO_VALIDATE:
+                    if (
+                        other_list_blocks_answer_ids[child_block]
+                        & list_blocks_answer_ids[child_block]
```

Comment:
Non blocker - since you used `.intersection` above instead of `&` I'd probably also do so here just for consistency

Reply:
Cool that makes sense, thanks